covidthailand.py

```diff
@@ -3467,156 +3876,208 @@ def save_plots(df):
+              kind='area', stacked=False, percent_fig=False, ma=False, cmap='tab20')
 
 
+    #######################
+    # Cases by provinces
+    #######################
     cases = pd.read_csv("api/cases_by_province.csv")
     cases['Date'] = pd.to_datetime(cases["Date"])
-    cases = cases.set_index(["Date","Province"])
+    cases = cases.set_index(["Date", "Province"])
 
-    def trendline(data, order=1):
-        # simulate dates with monotpnic inc numbers
-        dates = range(0,len(data.index.values))
+    def trendline(data: pd.DataFrame, order: int =1) -> float:
+        # simulate dates with monotonic inc numbers
+        dates = range(0, len(data.index.values))
         coeffs = np.polyfit(dates, list(data), order)
         slope = coeffs[-2]
         return float(slope)
 
-    increasing = lambda adf: adf["Cases"].rolling(3).mean().rolling(3).apply(trendline)
-    casesma = lambda adf: adf["Cases"].rolling(3).mean()
-    top5 = cases.pipe(topprov, increasing, casesma, name="Province Cases (3d MA)", other_name=None, num=5)
-    fig, ax = plt.subplots(figsize=[20, 10])
-    top5.last("30d").plot.line(
-        ax=ax,
-        #stacked=False,
-        title="Provinces with Cases Trending Up\n"
-        "in last 3 days (using 3 days rolling average)\n"
-        f"Updated: {TODAY().date()}\n"
-        "djay.github.io/covidthailand",
-    )
-    plt.tight_layout()
-    plt.savefig("outputs/cases_prov_increasing.png")
+    def increasing(adf: pd.DataFrame) -> pd.DataFrame:
+        return adf["Cases"].rolling(3).mean().rolling(3).apply(trendline)
 
-    decreasing = lambda adf: 1/increasing(adf)
-    top5 = cases.pipe(topprov, decreasing, casesma, name="Province Cases (3d MA)", other_name=None, num=5)
-    fig, ax = plt.subplots(figsize=[20, 10])
-    top5.last("30d").plot.line(
-        ax=ax,
-        #stacked=False,
-        title="Provinces with Cases Trending Down\n"
-        " in the last 3 days (using 3 days rolling average)\n"
-        f"Updated: {TODAY().date()}\n"
-        "djay.github.io/covidthailand",
-    )
-    plt.tight_layout()
-    plt.savefig("outputs/cases_prov_decreasing.png")
+    def cases_ma(adf: pd.DataFrame) -> pd.DataFrame:
+        return adf["Cases"].rolling(3).mean()
+
+    def decreasing(adf: pd.DataFrame) -> pd.DataFrame:
+        return 1/increasing(adf)
+
+    top5 = cases.pipe(topprov, increasing, cases_ma, name="Province Cases (3d MA)", other_name=None, num=5)
+    # fig, ax = plt.subplots(figsize=[20, 10])
+    # top5.last("30d").plot.line(
+    #     ax=ax,
+    #     # stacked=False,
+    #     title="Provinces with Cases Trending Up\n"
+    #     "in last 30 days (using 3 days rolling average)\n"
+    #     f"Updated: {TODAY().date()}\n"
+    #     "djay.github.io/covidthailand",
+    # )
+    # plt.tight_layout()
+    # plt.savefig("outputs/cases_prov_increasing.png")
+    cols = top5.columns.to_list()
+    plot_area(df=top5.last('30d'), png_prefix='cases_prov_increasing', cols_subset=cols,
```

Comment:
This is another special time period. so doesn't need it generated for all time periods?
or we add in a _30d (or _m) to all the other plots?

Reply:
We can either 1) filter the DataFrame before it is being passed to the `plot_area` function (such as in the example above) or pass the entire DataFrame and select the `_30d` plot. Which one do you favour?

Reply:
`_30d`